gwtp-core/gwtp-dispatch-rest/src/main/resources/com/gwtplatform/dispatch/rest/rebind/action/Action.vm

```diff
@@ -19,13 +22,40 @@ public class $impl extends AbstractRestAction<$result> {
 #if($bodyParameterName)
         setBodyParam($bodyParameterName);
 #end
-#if($contentType)
-        addParam(Type.HEADER, HttpHeaders.CONTENT_TYPE, "$contentType");
-#end
     }
 
     @Override
     public boolean isSecured() {
         return $secured;
     }
+
+    @Override
+    public String getBodyClass() {
+        return #if($body)"$body"#{else}null#end;
```

Comment:
Did you try to simply use : `return $body;`. If I recall, velocity ouputs null if the variable is null